src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -48,4 +52,18 @@ final public function isGranted(mixed $attribute, mixed $subject = null, ?Access
             array_pop($this->accessDecisionStack);
         }
     }
+
+    final public function isGrantedForUser(UserInterface $user, mixed $attribute, mixed $subject = null, ?AccessDecision $accessDecision = null): bool
+    {
+        $token = new class($user->getRoles()) extends AbstractToken implements OfflineTokenInterface {};
+        $token->setUser($user);
+        $this->tokenStack[] = $token;
+
+        try {
+            return $this->isGranted($attribute, $subject, $accessDecision);
+        } finally {
+            array_pop($this->tokenStack);
+            array_pop($this->accessDecisionStack);
```

Comment:
this array_pop is wrong. `isGranted` is already popping the access decision is is adding in the stack. This would pop a parent one.